clients/drcachesim/tracer/drmemtrace.h

```diff
@@ -149,6 +169,18 @@ drmemtrace_replace_file_ops(drmemtrace_open_file_func_t open_file_func,
                             drmemtrace_close_file_func_t close_file_func,
                             drmemtrace_create_dir_func_t create_dir_func);
 
+DR_EXPORT
+/**
+ * Identical to drmemtrace_replace_file_ops() except its file open function takes
+ * two extra parameters.
+ */
+drmemtrace_status_t
+drmemtrace_replace_file_ops_ex(drmemtrace_open_file_ex_func_t open_file_ex_func,
+                               drmemtrace_read_file_func_t read_file_func,
```

Comment:
Perhaps we should change the args to this routine to a struct (file_ops_func_t itself)? (to avoid an _ex_ex in future).

Reply:
+1, done.  The header is extern C though so it has to be a new struct.